lib/WebSocketServer.js

```diff
@@ -256,7 +256,8 @@ function handleHybiUpgrade(req, socket, upgradeHead, cb) {
     }
 
     // allows external modification/inspection of handshake headers
-    self.emit('headers', headers);
+    self.emit('headers' + req.url, headers, req.headers);
+    self.emit('headers', headers, req.headers);
```

Comment:
Instead of using two different events, why not using `req` as 3rd argument?

```js
this.emit('headers', headers, req);
```

so you can get all the required info (including the URL) from the `req` object if needed.